File: WPSKitTests/UIColorTests.m
//
//  UIColorTests.m
//  WPSKitTests
//
//  Created by Kirby Turner on 11/22/13.
//  Copyright (c) 2013 White Peak Software Inc. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "UIColor+WPSKit.h"

@interface UIColorTests : XCTestCase

@end

@implementation UIColorTests

- (void)setUp
{
  [super setUp];
  // Put setup code here; it will be run once, before the first test case.
}

- (void)tearDown
{
  // Put teardown code here; it will be run once, after the last test case.
  [super tearDown];
}

- (void)testColorWithHex
{
  UIColor *color = [UIColor wps_colorWithHex:0x007aff];
  NSString *hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"007AFF"], @"Unexpected hex string value (%@)", hexString);
  
  color = [UIColor wps_colorWithHex:0xe3e3e3];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"E3E3E3"], @"Unexpected hex string value (%@)", hexString);
  
  color = [UIColor wps_colorWithHex:0xAAAAAA];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"AAAAAA"], @"Unexpected hex string value (%@)", hexString);
}

- (void)testColorWithHexString
{
  UIColor *color = [UIColor wps_colorWithHexString:@"#007AFF"];
  NSString *hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"007AFF"], @"Unexpected hex string value (%@)", hexString);

  color = [UIColor wps_colorWithHexString:@"e3e3e3"];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"E3E3E3"], @"Unexpected hex string value (%@)", hexString);

  color = [UIColor whiteColor];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"FFFFFF"], @"Unexpected hex string value (%@)", hexString);
  
  color = [UIColor yellowColor];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"FFFF00"], @"Unexpected hex string value (%@)", hexString);

  color = [UIColor redColor];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"FF0000"], @"Unexpected hex string value (%@)", hexString);

  color = [UIColor lightGrayColor];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"AAAAAA"], @"Unexpected hex string value (%@)", hexString);

  color = [UIColor blackColor];
  hexString = [color wps_hexString];
  XCTAssertTrue([hexString isEqualToString:@"000000"], @"Unexpected hex string value (%@)", hexString);
}

@end

File: WPSKitTests/UIKit/UIColorTests.swift
//
//  UIColorTests.swift
//  WPSKit
//
//  Created by Kirby Turner on 3/5/16.
//  Copyright © 2016 White Peak Software, Inc. All rights reserved.
//

import XCTest

class UIColorTests: XCTestCase {

  func testColorWithHex() {
    var color = UIColor(hex: 0x007aff)
    XCTAssertEqual(color.hexString, "007AFF")
    
    color = UIColor(hex: 0xe3e3e3)
    XCTAssertEqual(color.hexString, "E3E3E3")
    
    color = UIColor(hex: 0xAAAAAA)
    XCTAssertEqual(color.hexString, "AAAAAA")
  }
  
  func testColorWithHexString() {
    var color = UIColor(hexString: "007AFF")
    XCTAssertEqual(color.hexString, "007AFF")

    color = UIColor(hexString: "e3e3e3")
    XCTAssertEqual(color.hexString, "E3E3E3")

    color = UIColor.whiteColor()
    XCTAssertEqual(color.hexString, "FFFFFF")

    color = UIColor.yellowColor()
    XCTAssertEqual(color.hexString, "FFFF00")

    color = UIColor.redColor()
    XCTAssertEqual(color.hexString, "FF0000")
    
    color = UIColor.lightGrayColor()
    XCTAssertEqual(color.hexString, "AAAAAA")

    color = UIColor.blackColor()
    XCTAssertEqual(color.hexString, "000000")
  }
  
  func testColors() {
    XCTAssertEqual(UIColor.iOSDefaultBlue().hexString, "007AFF")
    XCTAssertEqual(UIColor.facebookBlue().hexString, "3B5999")
    XCTAssertEqual(UIColor.twitterBlue().hexString, "55ACEE")

    XCTAssertNotNil(UIColor.randomColor())
  }
}
